Build is failing on macOS (FLEX_INCLUDE_DIR is set to NOTFOUND)

Hey,
when I try to bo build retdec, it fails during the compilation of `yaramod`

build/external/src/yaramod-project-stamp/yaramod-project-configure-err.log:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/Users/thau/tmp/retdec/build/external/src/yaramod-project/src/FLEX_INCLUDE_DIR
   used as include directory in directory /Users/thau/tmp/retdec/build/external/src/yaramod-project/src
```

build/external/src/yaramod-project-stamp/yaramod-project-configure-out.log
```
-- The CXX compiler identification is AppleClang 9.0.0.9000039
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found FLEX: /Users/thau/.local/opt/flex/bin/flex (found version "2.6.4")
-- Found BISON: /Users/thau/.local/opt/bison/bin/bison (found version "3.0.4")
-- Configuring incomplete, errors occurred!
See also "/Users/thau/tmp/retdec/build/external/src/yaramod-project-build/CMakeFiles/CMakeOutput.log".
```

Console output:
```
Scanning dependencies of target yaramod-project
[  8%] Creating directories for 'yaramod-project'
[  8%] Performing download step (download, verify and extract) for 'yaramod-project'
-- yaramod-project download command succeeded.  See also /Users/thau/tmp/retdec/build/external/src/yaramod-project-stamp/yaramod-project-download-*.log
[  8%] No patch step for 'yaramod-project'
[  8%] No update step for 'yaramod-project'
[  8%] Performing configure step for 'yaramod-project'
CMake Error at /Users/thau/tmp/retdec/build/external/src/yaramod-project-stamp/yaramod-project-configure-Release.cmake:16 (message):
  Command failed: 1

   '/Users/thau/.local/Cellar/cmake/3.10.1/bin/cmake' '-DCMAKE_BUILD_TYPE=Release' '-DCMAKE_C_COMPILER=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc' '-DCMAKE_CXX_COMPILER=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++' '-GUnix Makefiles' '/Users/kmadel/tmp/retdec/build/external/src/yaramod-project'

  See also

    /Users/thau/tmp/retdec/build/external/src/yaramod-project-stamp/yaramod-project-configure-*.log


make[2]: *** [external/src/yaramod-project-stamp/yaramod-project-configure] Error 1
make[1]: *** [deps/yaramod/CMakeFiles/yaramod-project.dir/all] Error 2
make: *** [all] Error 2
```

I was getting the same "FLEX_INCLUDE_DIR is set to NOTFOUND" error (but that ended up to be user error), but I'm now getting this error:

    parser/yy/parser.y:7.1-5: invalid directive: `%code'
    parser/yy/parser.y:7.7-14: syntax error, unexpected identifier
    make[5]: *** [src/yy_parser.cpp] Error 1
    make[4]: *** [src/CMakeFiles/yaramod.dir/all] Error 2
    make[3]: *** [all] Error 2

Seems kind of related (I'm on macOS 10.13.3), but if it needs to be reported as a separate issue, I'll do so.

I have meet this problem too...
parser/yy/parser.y:7.1-5: invalid directive: `%code'
parser/yy/parser.y:7.7-14: syntax error, unexpected identifier
make[5]: *** [src/yaramod/yy/yy_parser.cpp] Error 1
make[4]: *** [src/CMakeFiles/yaramod.dir/all] Error 2
make[3]: *** [all] Error 2

Zzm317-MacBook-Pro:~ zzm317$ cd downloads
Zzm317-MacBook-Pro:downloads zzm317$ cd retdec
Zzm317-MacBook-Pro:retdec zzm317$ cd build
Zzm317-MacBook-Pro:build zzm317$ ls
CMakeCache.txt		cmake_install.cmake	scripts
CMakeFiles		deps			src
Makefile		external
Zzm317-MacBook-Pro:build zzm317$ make install
[  0%] Built target capstone-project
[  1%] Built target jsoncpp-project
[  2%] Built target llvm-project
[  3%] Built target openssl
[  4%] Built target rapidjson-project
[  5%] Built target tinyxml2-project
[  6%] Built target elfio-project
[  7%] Built target libdwarf-project
[  8%] Built target pelib-project
[  9%] Built target yaracpp-project
[ 10%] Performing build step for 'yaramod-project'
CMake Error at /Users/zzm317/downloads/retdec/build/external/src/yaramod-project-stamp/yaramod-project-build-Release.cmake:16 (message):
  Command failed: 2

   '/Applications/Xcode.app/Contents/Developer/usr/bin/make'

  See also

    /Users/zzm317/downloads/retdec/build/external/src/yaramod-project-stamp/yaramod-project-build-*.log


make[2]: *** [external/src/yaramod-project-stamp/yaramod-project-build] Error 1
make[1]: *** [deps/yaramod/CMakeFiles/yaramod-project.dir/all] Error 2
make: *** [all] Error 2
Zzm317-MacBook-Pro:build zzm317$ make clean

I am having a similar problem when trying to build retdec for Windows 7 64bit.
Here is the fragment of the output of build process
15:21:04.470     1>
14>                     CMake Error at D:/retDec_newInstall/external/src/yaramod-project-stamp/yaramod-project-configure-Release.cmake:16 (message): (TaskId:40)
16>Target "AfterBuild: (TargetId:372)" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.BuildSteps.Targets" from project "D:\retDec_newInstall\src\crypto\retdec-crypto.vcxproj" (target "Build" depends on it):
14>                       Command failed: 1 (TaskId:40)
16>15:21:04.470     1>
14>                      (TaskId:40)
16>Done building target "AfterBuild" in project "retdec-crypto.vcxproj".: (TargetId:372)
14>                        'D:/Program Files/CMake/bin/cmake.exe' '-DCMAKE_BUILD_TYPE=Release' '-DCMAKE_C_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe' '-DCMAKE_CXX_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe' '-GVisual Studio 14 2015 Win64' 'D:/retDec_newInstall/external/src/yaramod-project' (TaskId:40)
16>15:21:04.470     1>
14>                      (TaskId:40)
16>Target "FinalizeBuildStatus: (TargetId:373)" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets" from project "D:\retDec_newInstall\src\crypto\retdec-crypto.vcxproj" (target "Build" depends on it):
14>                       See also (TaskId:40)
16>                   Task "Delete" (TaskId:135)
14>                      (TaskId:40)
16>                     Task Parameter:Files=retdec-crypto.dir\Release\retdec-crypto.tlog\unsuccessfulbuild (TaskId:135)
14>                         D:/retDec_newInstall/external/src/yaramod-project-stamp/yaramod-project-configure-*.log (TaskId:40)
14>                      (TaskId:40)
14>                      (TaskId:40)
14>15:21:08.600     1>
14>C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 1.
14>                   Done executing task "CustomBuild" -- FAILED. (TaskId:40)
14>15:21:08.601     1>
14>Done building target "CustomBuild" in project "yaramod-project.vcxproj" -- FAILED.: (TargetId:82)

The problems seems to be with yaramod library not being build successfully. I tried both debug and release configuration, but pretty much with the same result. 
Here is the list of my variables:
CMAKE_INCLUDE_PATH = D:/Program Files (x86)/GnuWin32/include 
CMAKE_LIBRARY_PATH = D:/Program Files (x86)/GnuWin32/lib;D:/flexbison/winflexbison/vs2015/release  
and this line was added to my PATH:
D:\win_flex_bison3-latest;D:/Program Files (x86)/GnuWin32/bin;D:/FlexBison/windlexbison/bin/release

I am very new to the projects of such complexity, CMAKE and MSBuild, so please do not be to harsh if I am not making too much sense. I would be happy to provide any additional info required to solve my problem.
 My ultimate goal is to build retdec and be able to step through its code because I think that would be a great learning experience.
Please help!

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -1,7 +1,6 @@
 version: '{build}'
 
 environment:
-  WINFLEXBISON_ARCHIVE: winflexbison-2.5.16.zip
   matrix:
     - CMAKE_GENERATOR: Visual Studio 15 2017 Win64
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
@@ -10,9 +9,6 @@ matrix:
   fast_finish: true
 
 install:
-  - appveyor DownloadFile "https://downloads.sourceforge.net/project/winflexbison/old_versions/%WINFLEXBISON_ARCHIVE%"
-  - 7z x -y -owinflexbison\ "%WINFLEXBISON_ARCHIVE%" > nul
-  - set Path=%CD%\winflexbison;%Path%
   - set Path=C:\Python36;%Path%
 
 before_build:

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,8 +12,6 @@ matrix:
           packages:
             - build-essential
             - gcc-multilib
-            - flex
-            - bison
             - autoconf
             - automake
             - libtool
@@ -34,8 +32,6 @@ matrix:
         - CCACHE_CPP2=true
 
 install:
-  # We need to install newer versions of Flex and Bison on MacOS X.
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install flex bison; fi
   # ccache is not installed on OS X.
   - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install ccache; fi
   # install for os x the gnu time library
@@ -43,10 +39,6 @@ install:
 
 before_script:
   - eval "${MATRIX_EVAL}"
-  # We need to use newer versions of Flex and Bison on MacOS X (the ones from Homebrew).
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then export CMAKE_INCLUDE_PATH="/usr/local/opt/flex/include"; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then export CMAKE_LIBRARY_PATH="/usr/local/opt/flex/lib;/usr/local/opt/bison/lib"; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then export PATH="/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH"; fi
   # We need to add ccache before everything else into PATH.
   - if [ "$TRAVIS_OS_NAME" = "osx" ]; then export PATH="/usr/local/opt/ccache/libexec:$PATH"; fi
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,7 @@
 * Enhancement: Added CMake options to build and install only specific targets ([#510](https://github.com/avast/retdec/issues/510)).
 * Enhancement: Switched from C++14 to C++17 ([#650](https://github.com/avast/retdec/issues/650)).
 * Enhancement: Replace uses of `mpark::variant` from `deps/variant` with standard C++17 `std::variant`. Remove the `variant` dependency.
+* Enhancement: Updated Yaramod to version v3.0.0 ([#680](https://github.com/avast/retdec/pull/680)). RetDec no longer requires Flex and Bison. This fixes [#103](https://github.com/avast/retdec/issues/103).
 * Fix: Increased the limit for the number of entries in import directory when deciding whether a PE file is corrupted or not ([avast/pelib#13](https://github.com/avast/pelib/pull/13)).
 * Fix: Fixed build on BSD systems ([#598](https://github.com/avast/retdec/pull/598)).
 * Fix: Resources which are located in the different section than resource tree are now properly parsed ([#596](https://github.com/avast/retdec/pull/596)).

--- a/Dockerfile
+++ b/Dockerfile
@@ -14,8 +14,6 @@ RUN apt-get -y update && \
 	doxygen                                             \
 	graphviz                                            \
 	upx                                                 \
-	flex                                                \
-	bison                                               \
 	openssl                                             \
 	libssl-dev                                          \
 	zlib1g-dev                                          \

--- a/Dockerfile.dev
+++ b/Dockerfile.dev
@@ -14,8 +14,6 @@ RUN apt-get -y update && \
 	doxygen                                             \
 	graphviz                                            \
 	upx                                                 \
-	flex                                                \
-	bison                                               \
 	openssl                                             \
 	libssl-dev                                          \
 	zlib1g-dev                                          \

--- a/README.md
+++ b/README.md
@@ -125,8 +125,6 @@ This section describes a local build and installation of RetDec. Instructions fo
 * [Git](https://git-scm.com/)
 * [Perl](https://www.perl.org/)
 * [Python](https://www.python.org/) (version >= 3.4)
-* [Bison](https://www.gnu.org/software/bison/) (version >= 3.0)
-* [Flex](https://www.gnu.org/software/flex/) (version >= 2.6)
 * [autotools](https://en.wikipedia.org/wiki/GNU_Build_System) ([autoconf](https://www.gnu.org/software/autoconf/autoconf.html), [automake](https://www.gnu.org/software/automake/), and [libtool](https://www.gnu.org/software/libtool/))
 * [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/)
 * [m4](https://www.gnu.org/software/m4/m4.html)
@@ -136,19 +134,19 @@ This section describes a local build and installation of RetDec. Instructions fo
 On Debian-based distributions (e.g. Ubuntu), the required packages can be installed with `apt-get`:
 
 ```sh
-sudo apt-get install build-essential cmake git perl python3 bison flex libfl-dev autoconf automake libtool pkg-config m4 zlib1g-dev upx doxygen graphviz
+sudo apt-get install build-essential cmake git perl python3 autoconf automake libtool pkg-config m4 zlib1g-dev upx doxygen graphviz
 ```
 
 On RPM-based distributions (e.g. Fedora), the required packages can be installed with `dnf`:
 
 ```sh
-sudo dnf install gcc gcc-c++ cmake make git perl python3 bison flex autoconf automake libtool pkg-config m4 zlib-devel upx doxygen graphviz
+sudo dnf install gcc gcc-c++ cmake make git perl python3 autoconf automake libtool pkg-config m4 zlib-devel upx doxygen graphviz
 ```
 
 On Arch Linux, the required packages can be installed with `pacman`:
 
 ```sh
-sudo pacman --needed -S base-devel cmake git perl python3 bison flex autoconf automake libtool pkg-config m4 zlib upx doxygen graphviz
+sudo pacman --needed -S base-devel cmake git perl python3 autoconf automake libtool pkg-config m4 zlib upx doxygen graphviz
 ```
 
 #### Windows
@@ -156,7 +154,6 @@ sudo pacman --needed -S base-devel cmake git perl python3 bison flex autoconf au
 * Microsoft Visual C++ (version >= Visual Studio 2017 version 15.7)
 * [CMake](https://cmake.org/) (version >= 3.6)
 * [Git](https://git-scm.com/)
-* [Flex + Bison](https://sourceforge.net/projects/winflexbison/files/win_flex_bison3-latest.zip/download) ([mirror](https://github.com/avast/retdec-support/releases/download/2018-07-27/win_flex_bison3-latest.zip)) from the [Win flex-bison project](https://sourceforge.net/projects/winflexbison/). Add the extracted directory to the system `Path` ([HOWTO](https://www.computerhope.com/issues/ch000549.htm)).
 * [Active Perl](https://www.activestate.com/activeperl). It needs to be the first Perl in `PATH`, or it has to be provided to CMake using `CMAKE_PROGRAM_PATH` variable, e.g. `-DCMAKE_PROGRAM_PATH=/c/perl/bin`. Does NOT work with Strawberry Perl or MSYS2 Perl (you would have to install a pre-built version of OpenSSL, see below).
   * Alternatively, you can install OpenSSL directly from [here](https://slproweb.com/products/Win32OpenSSL.html). This means OpenSSL won't be built and you don't need to install any Perl. Do not install Light version of OpenSSL as they don't contain development files.
 * [Python](https://www.python.org/) (version >= 3.4)
@@ -172,8 +169,6 @@ Packages should be preferably installed via [Homebrew](https://brew.sh).
 * [Git](https://git-scm.com/)
 * [Perl](https://www.perl.org/)
 * [Python](https://www.python.org/) (version >= 3.4)
-* [Bison](https://www.gnu.org/software/bison/) (version >= 3.0)
-* [Flex](https://www.gnu.org/software/flex/) (version >= 2.6)
 * [autotools](https://en.wikipedia.org/wiki/GNU_Build_System) ([autoconf](https://www.gnu.org/software/autoconf/autoconf.html), [automake](https://www.gnu.org/software/automake/), and [libtool](https://www.gnu.org/software/libtool/))
 * Optional: [Doxygen](http://www.stack.nl/~dimitri/doxygen/) and [Graphviz](http://www.graphviz.org/) for generating API documentation
 
@@ -182,7 +177,7 @@ Packages should be preferably installed via [Homebrew](https://brew.sh).
 Packages should be installed via FreeBSDs pre-compiled package repository using the `pkg` command or built from scratch using the `ports` database method.
 
 * Full "pkg" tool instructions: [handbook pkg method](https://www.freebsd.org/doc/handbook/pkgng-intro.html)
-  * `pkg install cmake python37 bison git autotools`
+  * `pkg install cmake python37 git autotools`
 OR
 * Full "ports" instructions: [handbook ports method](https://www.freebsd.org/doc/handbook/ports-using.html)
   * `portsnap fetch`
@@ -215,12 +210,6 @@ Note: Although RetDec now supports a system-wide installation ([#94](https://git
 * macOS:
   * `cd retdec`
   * `mkdir build && cd build`
-  * ```sh
-    # Apple ships old Flex & Bison, so Homebrew versions should be used.
-    export CMAKE_INCLUDE_PATH="/usr/local/opt/flex/include"
-    export CMAKE_LIBRARY_PATH="/usr/local/opt/flex/lib;/usr/local/opt/bison/lib"
-    export PATH="/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH"
-    ```
   * `cmake .. -DCMAKE_INSTALL_PREFIX=<path>`
   * `make -jN` (`N` is the number of processes to use for parallel build, typically number of cores + 1 gives fastest compilation time)
   * `make install`
@@ -230,12 +219,12 @@ Note: Although RetDec now supports a system-wide installation ([#94](https://git
   * `cd retdec`
   * `mkdir build && cd build`
   * ```sh
-    # FreeBSD (and other BSDs) do need cmake, python3, bison, git, autotools. Flex and perl are pre-installed in the OS but check versions.
+    # FreeBSD (and other BSDs) do need cmake, python3, git, autotools. Perl is pre-installed in the OS but check its version.
     # Later versions may be available for each of the packages.
     # See what is installed:
-    sudo pkg info cmake python37 bison autotools
+    sudo pkg info cmake python37 autotools
     # Install/upgrade them:
-    sudo pkg install cmake python37 bison autotools
+    sudo pkg install cmake python37 autotools
     ```
   * `cmake .. -DCMAKE_INSTALL_PREFIX=<path>`
   * `make -jN` (`N` is the number of processes to use for parallel build, typically number of cores + 1 gives fastest compilation time)

--- a/deps/yaramod/CMakeLists.txt
+++ b/deps/yaramod/CMakeLists.txt
@@ -1,7 +1,3 @@
-# Requirements.
-find_package(FLEX REQUIRED)
-find_package(BISON REQUIRED)
-
 include(ExternalProject)
 
 if(CMAKE_C_COMPILER)

--- a/deps/yaramod/CMakeLists.txt
+++ b/deps/yaramod/CMakeLists.txt
@@ -35,8 +35,8 @@ else()
 	message(STATUS "YaraMod: using remote YaraMod revision.")
 
 	ExternalProject_Add(yaramod-project
-		URL https://github.com/avast/yaramod/archive/v2.12.1.zip
-		URL_HASH SHA256=16414f53ed8480e848dad665779857f3f7159cdf995f2e69d6539247ee2dbc2e
+		URL https://github.com/avast/yaramod/archive/v3.0.0b0.zip
+		URL_HASH SHA256=21b83214e72dccc4569183c30025e054fc372d0ebe0de2c909ffdad3a49786f4
 		DOWNLOAD_NAME yaramod.zip
 		CMAKE_ARGS
 			# This does not work on MSVC, but may be useful on Linux.
@@ -72,6 +72,17 @@ add_dependencies(yaramod yaramod-project)
 target_include_directories(yaramod SYSTEM INTERFACE ${source_dir}/include)
 target_include_directories(yaramod SYSTEM INTERFACE ${source_dir}/deps)
 target_include_directories(yaramod SYSTEM INTERFACE ${binary_dir}/src)
-target_include_directories(yaramod SYSTEM INTERFACE "${FLEX_INCLUDE_DIRS}")
-target_link_libraries(yaramod INTERFACE debug     ${binary_dir}/src/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}yaramod${CMAKE_STATIC_LIBRARY_SUFFIX})
-target_link_libraries(yaramod INTERFACE optimized ${binary_dir}/src/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}yaramod${CMAKE_STATIC_LIBRARY_SUFFIX})
+target_include_directories(yaramod SYSTEM INTERFACE ${binary_dir}/deps/pog/pog-install/include)
+target_include_directories(yaramod SYSTEM INTERFACE ${binary_dir}/deps/pog/pog-install/include/pog)
+target_link_libraries(yaramod
+	INTERFACE debug
+		${binary_dir}/src/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}yaramod${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_fmt${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_re2${CMAKE_STATIC_LIBRARY_SUFFIX}
+)
+target_link_libraries(yaramod
+	INTERFACE optimized
+		${binary_dir}/src/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}yaramod${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_fmt${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_re2${CMAKE_STATIC_LIBRARY_SUFFIX}
+)

--- a/deps/yaramod/CMakeLists.txt
+++ b/deps/yaramod/CMakeLists.txt
@@ -73,12 +73,12 @@ target_include_directories(yaramod SYSTEM INTERFACE ${binary_dir}/deps/pog/pog-i
 target_link_libraries(yaramod
 	INTERFACE debug
 		${binary_dir}/src/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}yaramod${CMAKE_STATIC_LIBRARY_SUFFIX}
-		${binary_dir}/deps/pog/pog-install/lib/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_fmt${CMAKE_STATIC_LIBRARY_SUFFIX}
-		${binary_dir}/deps/pog/pog-install/lib/${DEBUG_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_re2${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}pog_fmt${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}pog_re2${CMAKE_STATIC_LIBRARY_SUFFIX}
 )
 target_link_libraries(yaramod
 	INTERFACE optimized
 		${binary_dir}/src/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}yaramod${CMAKE_STATIC_LIBRARY_SUFFIX}
-		${binary_dir}/deps/pog/pog-install/lib/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_fmt${CMAKE_STATIC_LIBRARY_SUFFIX}
-		${binary_dir}/deps/pog/pog-install/lib/${RELEASE_DIR}${CMAKE_STATIC_LIBRARY_PREFIX}pog_re2${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}pog_fmt${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${binary_dir}/deps/pog/pog-install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}pog_re2${CMAKE_STATIC_LIBRARY_SUFFIX}
 )

--- a/src/pat2yara/modifications.cpp
+++ b/src/pat2yara/modifications.cpp
@@ -154,7 +154,9 @@ std::shared_ptr<HexString> cutHexString(
 			old.begin() + nibbleLimit);
 
 		// Create new hexadecimal string.
-		return std::make_shared<HexString>(newUnits);
+		return std::make_shared<HexString>(
+				hexString->getTokenStream(),
+				newUnits);
 	}
 
 	// String is already shorter than limit.

--- a/src/pat2yara/processing.cpp
+++ b/src/pat2yara/processing.cpp
@@ -159,10 +159,12 @@ void processFiles(
 	bool firstFile = true;
 	std::vector<std::unique_ptr<Rule>> rules;
 
+	Yaramod ym;
+
 	std::size_t counter = 0;
 	for (const auto &file : options.input) {
 		// Parse file.
-		auto yaraFile = parseFile(file);
+		auto yaraFile = ym.parseFile(file);
 
 		// Add architecture info rule.
 		if (firstFile) {

